AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -508,4 +508,17 @@ - (NSArray *)nodesAtIndexPaths:(NSArray *)indexPaths
   return ASFindElementsInMultidimensionalArrayAtIndexPaths(_nodes, [indexPaths sortedArrayUsingSelector:@selector(compare:)]);
 }
 
+#pragma mark - Dealloc
+
+- (void)dealloc {
+  ASDisplayNodeAssertMainThread();
+  [_nodes enumerateObjectsUsingBlock:^(NSMutableArray *section, NSUInteger sectionIndex, BOOL *stop) {
+    [section enumerateObjectsUsingBlock:^(ASCellNode *node, NSUInteger rowIndex, BOOL *stop) {
+      if (node.isNodeLoaded && node.view.superview) {
```

Comment:
Definitely correct form to check .isNodeLoaded here (I'm sure this is obvious to you, but it's nice to see a well considered patch :) )

I would remove the second part of the condition, as it performs two method calls.  Additionally, it's not compatible with layer-backing.